core/commands/store/contracts/contracts.go

```diff
@@ -193,25 +200,37 @@ This command get contracts list based on role from the local node data store.`,
 		if !ok {
 			return fmt.Errorf("invalid filter option: %s", filterOpt)
 		}
-		// TODO: Use size
-		// TODO: sort
-		//sizeOpt, _ := req.Options[contractsListSizeOptionName].(int)
-		contracts, err := ListContracts(n.Repo.Datastore(), req.Arguments[0])
+		size, ok := req.Options[contractsListSizeOptionName].(int)
+		if !ok {
+			return fmt.Errorf("invalid size option: %s", size)
+		}
+		contracts, err := ListContracts(n.Repo.Datastore(), cr.String())
 		if err != nil {
 			return err
 		}
+		sort.Sort(ByTime(contracts))
+		if parts[1] == "" || parts[1] == "desc" {
+			// reverse
+			for i, j := 0, len(contracts)-1; i < j; i, j = i+1, j-1 {
+				contracts[i], contracts[j] = contracts[j], contracts[i]
+			}
+		}
 		result := make([]*nodepb.Contracts_Contract, 0)
 		for _, c := range contracts {
 			b := false
 			for _, state := range states {
```

Comment:
Maybe should make `states` a map.. when i created it i didn't know the actual usage pattern..

Reply:
i tried to use a set/map, but `guardpb.Contract_DRAFT` can't be a key of map.